scanner/updater/export.go

```diff
@@ -0,0 +1,101 @@
+package updater
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/klauspost/compress/zstd"
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/libvuln/jsonblob"
+	"github.com/quay/claircore/libvuln/updates"
+	"github.com/quay/zlog"
+	"github.com/stackrox/rox/scanner/updater/manual"
+	"golang.org/x/time/rate"
+
+	_ "github.com/quay/claircore/updater/defaults"
+)
+
+// Export is responsible for triggering the updaters to download Common Vulnerabilities and Exposures (CVEs) data
+// and then outputting the result as a zstd-compressed file with .ztd extension
+func Export(ctx context.Context, outputDir string) error {
+
+	err := os.MkdirAll(outputDir, 0700)
+	if err != nil {
+		return err
+	}
+
+	// create output json file
+	outputFile, err := os.Create(filepath.Join(outputDir, "output.json.zst"))
+	if err != nil {
+		return err
+	}
+
+	limiter := rate.NewLimiter(rate.Every(time.Second), 5)
+	httpClient := &http.Client{
+		Transport: &rateLimitedTransport{
+			limiter:   limiter,
+			transport: http.DefaultTransport,
+		},
+	}
+
+	zstdWriter, err := zstd.NewWriter(outputFile)
+	if err != nil {
+		return err
+	}
+	defer zstdWriter.Close()
+
+	//add out of tree updater
+	updaterSet, err := manual.UpdaterSet(ctx, nil)
```

Comment:
Is your intention to continue even if the manual updater set creation failed (i.e. `err != nil`)?

Reply:
Yeah, if there's an error, we will continue with other data sources without manual input vulns.

Reply:
Could you explain why we should continue after manual CVEs fail? Also, how will we receive notifications of failures? Finally, will the bundle have no manual vulnerabilities at the end or just the new ones will be absent?

Reply:
Actually I should stop the process of updating vulns once manual input update fails because we have `fail-fast: false` so the failure only impacts updating one supported version and we can still get notified.